rust/kernel/error.rs

```diff
@@ -57,14 +57,34 @@ impl Error {
     pub const EBADF: Self = Error(-(bindings::EBADF as i32));
 
     /// Creates an [`Error`] from a kernel error code.
+    /// when errno given is invalid,
+    /// it will convert it to EINVAL, and print a warning message.
     pub fn from_kernel_errno(errno: c_types::c_int) -> Error {
+        Error::new(errno)
+    }
+
+    /// Creates an [`Error`] from a kernel error code without a sanity check
+    pub fn from_kernel_errno_unchecked(errno: c_types::c_int) -> Error {
         Error(errno)
     }
 
     /// Returns the kernel error code.
     pub fn to_kernel_errno(self) -> c_types::c_int {
         self.0
     }
+
+    /// INVARIANT: make sure Error is initialized with a sane value
+    /// When an invalid errno is found, it will
+    /// 1) convert it to EINVAL
+    /// 2) print a warning message
```

Comment:
Maybe swap the order of these two statements? Printing the warning happens before returning `EINVAL`.

Reply:
ok